pkg/cert/truststore.go

```diff
@@ -22,6 +23,10 @@ type TrustStore interface {
 	Verifier
 	Pool() *x509.CertPool
 	AddCertificate(certificate *x509.Certificate) error
+	// GetRoots returns all roots active at the given time
+	GetRoots(time.Time) []*x509.Certificate
+	// GetCertificates returns all certificates signed by given signer chains, active at the given time and if it must be a CA
```

Comment:
I don't understand the "and if it must be a CA" part

Reply:
For the certificate list in reverse-proxies you only want CA's? This filters all non-CA's